File: app/src/coldbrew/player/player.py
from random import random

class BCAbstractRobot:
	def __init__(self):
		self.game_state = None

	def _do_turn(self, game_state):
		self.game_state = game_state
        return self.turn()

    def move(self):
    	return None

    def turn(self):
    	return None

################################

class MyRobot(BCAbstractRobot):
	def turn(self):
		dir = int(random()*4)
		self.move(dir)

###############################

robot = MyRobot()

"""
def turn():
    dir = int(random()*4)
    while bc.inDirection(dir) is None:
        dir = int(random()*4)

    if random() > 0.9:
        bc.log("I exist!")

    if bc.inDirection(dir) > 0:
        return bc.attack(dir)
    else:
        return bc.move(dir)
"""

File: app/src/coldbrew/player/player.js
var bc = {
    NORTH: 0
}

class BCAbstractRobot {
    constructor() {
        this.game_state = null;
    }

    _do_turn(game_state) {
        this.game_state = game_state;
        return this.turn();   
    }

    move() {
        return null;
    }

    turn() {
        return null;
    }
}

//////////////////////////////////

class MyRobot extends BCRobot {
    turn() {
        return this.move(bc.NORTH);
    }
}

/////////////////////////////////

var robot = {'robot':MyRobot()};
/*
function turn() {
    var dir = 0;

    do {
        dir = Math.floor(Math.random()*4);
    } while (bc.inDirection(dir) == null);


    if (bc.inDirection(dir) == 0) {
        return bc.move(dir);
    } else {
        var robot = bc.getRobot(bc.inDirection(dir));
        if (robot.team == bc.me().team) return;
        if (robot.health < bc.me().health*0.4) {
            bc.log("time left: " + bc.timeLeft());
            return bc.split(dir);
        } else {
            return bc.attack(dir);
        }
    }
    return bc.move(bc.DOWN);
}
*/